### PR TITLE
Suresh/HTS-56 :  Get Applicant's Email Address

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosavestub/config/UserDetailsConnector.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+@Singleton
+class UserDetailsConnectorImpl extends UserDetailsConnector with ServicesConfig {
+
+  private val userDetailsRoot = baseUrl("user-details")
+  private val serviceURL = "user-details/id"
+  private val http = WSHttp
+
+  case class EmailException(id: String) extends Exception(s"Could not find email in the user details for id: $id ")
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavestub/config/UserDetailsConnector.scala]: Probably make move this into a companion object so it's visible outside of the class?

File: app/uk/gov/hmrc/helptosavestub/config/UserDetailsConnector.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  override def getEmail(userDetailsId: String)(implicit hc: HeaderCarrier): Future[Email] = {
+
+    http.GET(s"$userDetailsRoot/$serviceURL/$userDetailsId").flatMap {
+      _.json.\("email").validate[String] match {
+        case JsSuccess(email, _) => Future.successful(Email(email))
+        case _ ⇒ Future.failed[Email](EmailException(userDetailsId))
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavestub/config/UserDetailsConnector.scala]: Do we want to include some reasons why from the JsFailure?

File: project/MicroServiceBuild.scala [hunk 3]
```diff
@@ -26,7 +23,8 @@ object MicroServiceBuild extends Build with MicroService {
     "uk.gov.hmrc" %% "hmrctest" % "2.3.0" % scope,
     "org.scalatest" %% "scalatest" % "2.2.6" % scope,
     "org.pegdown" % "pegdown" % "1.6.0" % scope,
-    "com.typesafe.play" %% "play-test" % PlayVersion.current % scope
+    "com.typesafe.play" %% "play-test" % PlayVersion.current % scope,
+    "com.github.tomakehurst" % "wiremock" % "2.5.1" % scope
```
Comment (AndyHWChung) [inline project/MicroServiceBuild.scala]: I think we need to agree on what mocking library to use across the projects - I've used ScalaMock in the frontend project but that can be refactored to use WireMock if it is better. It'll be nice if we had consistency  across the projects

File: app/uk/gov/hmrc/helptosavestub/config/UserDetailsConnector.scala [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  private val userDetailsRoot = baseUrl("user-details")
+  private val serviceURL = "user-details/id"
+  private val http = WSHttp
+
+  case class EmailException(id: String) extends Exception(s"Could not find email in the user details for id: $id ")
+
+  override def getEmail(userDetailsId: String)(implicit hc: HeaderCarrier): Future[Email] = {
+
+    http.GET(s"$userDetailsRoot/$serviceURL/$userDetailsId").flatMap {
+      _.json.\("email").validate[String] match {
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavestub/config/UserDetailsConnector.scala]: can be shortened to `_.json.validate[Email]`

**Comment by sureshhmrc** [comment]
closing this PR as these changes will be moved to `eligibility-check` service

**Pull request closed without merging**